src/client/microsoftAuth.js

```diff
@@ -11,113 +10,61 @@ const getFetchOptions = {
   }
 }
 
-/**
- * Obtains Minecaft profile data using a Minecraft access token and starts the join sequence
- * @param {object} client - The client passed to protocol
- * @param {object} options - Client Options
- * @param {string} mcAccessToken - Minecraft access token for session server
- * @param {object?} msa - Cached Microsoft account data for more descriptive errors
- */
-async function postAuthenticate (client, options, mcAccessToken, msa) {
-  options.haveCredentials = mcAccessToken != null
-
-  let minecraftProfile
-  const res = await fetch(authConstants.MinecraftServicesProfile, getFetchOptions)
-  if (res.ok) { // res.status >= 200 && res.status < 300
-    minecraftProfile = await res.json()
-  } else {
-    const user = msa ? msa.getUsers()[0] : options.username
-    throw Error(`Failed to obtain Minecraft profile data for '${user?.username}', does the account own Minecraft Java? Server returned: ${res.statusText}`)
+async function authenticate (client, options) {
+  if (!options.profilesFolder && options.profilesFolder !== false) {
+    options.profilesFolder = require('minecraft-folder-path')
   }
 
-  if (!minecraftProfile.id) {
-    debug('[mc] profile', minecraftProfile)
+  const Authflow = new PrismarineAuth(options.username, options.profilesFolder, options, options.onMsaCode)
+  const accessToken = await Authflow.getMinecraftJavaToken()
+
+  // We have a minecraft access token.
+  // Let's verify entitlements.
+
+  getFetchOptions.headers.Authorization = `Bearer ${accessToken}`
+  const MineEntitlements = await fetch(Endpoints.MinecraftServicesEntitlement, getFetchOptions).then(checkStatus)
+  if (MineEntitlements.items.length === 0) throw Error('This user does not possess any entitlements on this account according to minecraft services.')
+
+  // We know that this account owns minecraft.
+  // Let's fetch the profile data.
+
+  options.haveCredentials = accessToken != null
+
+  const MinecraftProfile = await fetch(Endpoints.MinecraftServicesProfile, getFetchOptions)
```

Comment:
endpoints should not need to be exposed, why not do this check in prismarine-auth?

Reply:
Reason being is that prismarine-auth handles authentication and caching tokens. I can merge this into prismarine-auth if you feel the need to have it there instead of here. This just checks for entitlements as you would know.

Reply:
ok, do you think it would make sense for others to have this information if they're also using minecraft auth ? That way people don't have to have copies of this in their code that need mc auth. I think it would be a helpful additional check, maybe optional option to the getMinecraftToken call ?

Reply:
I'm not sure what you're getting at here The only thing I can think of is an extra check to see if you're actually entitled to the game or not